services/121-service/src/payments/fsp-integration/airtel/src/util.ts

```diff
@@ -0,0 +1,187 @@
+import { Buffer } from 'node:buffer';
+import { publicEncrypt } from 'node:crypto';
+const { constants } = await import('node:crypto');
+
+const rsaPublicKeyToPem = (key: string): string => {
+  const formattedKey = `-----BEGIN PUBLIC KEY-----\n${key
+    .match(/.{1,64}/g)
+    ?.join('\n')}\n-----END PUBLIC KEY-----`;
+  return formattedKey;
+};
+
+export const encryptPinV1 = (data: string, base64PublicKey: string): string => {
+  const publicKey = rsaPublicKeyToPem(base64PublicKey);
+  const encrypted = publicEncrypt(
+    {
+      key: publicKey,
+      padding: constants.RSA_PKCS1_PADDING,
+      oaepHash: 'sha256',
+    },
+    Buffer.from(data),
+  );
+  return encrypted.toString('base64');
+};
+
+export const getAccessToken = async ({
+  url,
+  client_id,
+  client_secret,
+}): Promise<string> => {
+  const headers = {
+    'Content-Type': 'application/json',
+    Accept: '*/*',
+  };
+
+  const body = JSON.stringify({
+    client_id,
+    client_secret,
+    grant_type: 'client_credentials',
+  });
+
+  try {
+    const response = await fetch(url, {
+      method: 'POST',
+      headers,
+      body,
+    });
+
+    if (!response.ok) {
+      console.log(response);
+      const errorDetails = await response.text();
+      throw new Error(
+        `Failed to fetch access token: ${response.status} ${response.statusText} - ${errorDetails}`,
+      );
+    }
+
+    const { access_token } = (await response.json()) as {
+      access_token: string;
+      expires_in: string;
+      token_type: string;
+    };
+    return access_token;
+  } catch (error) {
+    console.error(
+      'Error fetching access token:',
+      error instanceof Error ? error.message : error,
+    );
+    throw error;
+  }
+};
+
+export const generateRandomId = (length = 40): string => {
```

Comment:
For implementation I suggest we generate a random UUID, to be consistent of how we also do this in other FSP integrations. So this function would not be needed in the actual implementation.